estrest/test/test_causal_model.py

```diff
@@ -32,9 +32,36 @@ def test_intervention(self):
 
     def test_invalid_intervention(self):
         m = CausalModel()
-        m.add('A', lambda vals: True)
+        m.add('A', lambda vals: True, [])
         self.assertRaises(Exception, lambda: m.intervene({'X': True}))
 
+    def test_w_projection_vars(self):
+        m = CausalModel()
+        m.add('A', lambda vals: True, [])
+        m.add('B', lambda vals: False, [])
+        m.add('C', lambda vals: vals['A'] and vals['B'], ['A', 'B'])
+        m.add('D', lambda vals: vals['C'] and True, ['C'])
+
+        wp = m._w_projection_vars('A', 'D')
+        self.assertEqual(wp, {'A', 'B', 'C', 'D'})
+
```

Comment:
Add another test to check whether some variables are being removed.

Reply:
Done. I used the SUZY_BILLY model. For checking Billy not being a cause, ST must be removed.